rero_ils/query.py

```diff
@@ -163,6 +164,43 @@ def loans_search_factory(self, search, query_parser=None):
     return search, urlkwargs
 
 
+def templates_search_factory(self, search, query_parser=None):
+    """Template search factory.
+
+    Restricts results to organisation level for librarian and sys_lib.
+    Restricts results to private templates for users with role librarian.
+    """
+    search, urlkwargs = search_factory(self, search)
+    if current_patron:
+        if current_patron.is_system_librarian:
+            search = search.filter(
+                'term', organisation__pid=current_patron.get_organisation(
+                    )['pid'])
```

Comment:
you could replace `current_patron.get_organisation()['pid']` by easier `current_organisation.pid`